internal/pkg/agent/install/uninstall.go

```diff
@@ -30,13 +30,35 @@ import (
 )
 
 // Uninstall uninstalls persistently Elastic Agent on the system.
-func Uninstall(cfgFile, topPath, uninstallToken string) error {
-	// uninstall the current service
+func Uninstall(cfgFile, topPath, uninstallToken string, force bool) error {
+	var log *logp.Logger
+	var err error
+
+	if force {
+		log, err = logger.NewWithLogpLevel("uninstall", logp.DebugLevel, false)
+
+	} else {
+		log, err = logger.NewWithLogpLevel("uninstall", logp.ErrorLevel, false)
+	}
+	if err != nil {
+		return fmt.Errorf("error setting logger: %w", err)
+	}
+
 	svc, err := newService(topPath)
 	if err != nil {
 		return err
 	}
-	status, _ := svc.Status()
+
+	status, err := svc.Status()
+	if err != nil {
+		log.Debugf("failed to get status for service (%s): %s", paths.ServiceName, err)
+		if !force {
```

Comment:
Now that I've seen this, I don't love it. Seems like it makes the code more convoluted without necessarily fixing a real problem, not sure the extra complexity is worth the "just in case" insurance.